src/inet/UDPEndPointImplLwIP.cpp

```diff
@@ -241,6 +241,18 @@ void UDPEndPointImplLwIP::CloseImpl()
         udp_remove(mUDP);
         mUDP              = nullptr;
         mLwIPEndPointType = LwIPEndPointType::Unknown;
+
+        // In case that there is a UDPEndPointImplLwIP::LwIPReceiveUDPMessage
+        // event pending in the event queue (SystemLayer::ScheduleLambda), we
+        // schedule a release call to the end of the queue, to ensure that the
+        // queued pointer to UDPEndPointImplLwIP is not dangling.
+        Retain();
+        CHIP_ERROR err = GetSystemLayer().ScheduleLambda([this] { Release(); });
+        if (err != CHIP_NO_ERROR)
+        {
+            // There is nothing we can do here, accept the chance of racing
+            Release();
+        }
```

Comment:
Why are we calling the same Release() twice with 2 ways 
the first one is a callback ?

Reply:
if `ScheduleLambda` fails, the lambda will not be called, we should manually call release here, otherwise the ep will be leaked.

Ideally we should use RAII, using a empty lambda capturing a unique ptr like object calling release when destructed. But out lambda implementation do not support non-trival object.

I will rework this part in post 1.0